src/main/java/io/cdap/plugin/ariba/source/AribaStructuredTransformer.java

```diff
@@ -0,0 +1,376 @@
+
+    String colValTrim = fieldValStr.trim();
+    switch (fieldType) {
+      case INT:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        recBuilder.set(fieldName, Integer.parseInt(colValTrim));
+        break;
+
+      case LONG:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        recBuilder.set(fieldName, Long.parseLong(colValTrim));
+        break;
+
+      case FLOAT:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        // FLOAT values can contain exponential values.
+        recBuilder.set(fieldName, new BigDecimal(colValTrim));
+        break;
+
+      case DOUBLE:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        recBuilder.set(fieldName, Double.parseDouble(colValTrim));
+        break;
+
+      case BYTES:
+        recBuilder.set(fieldName, Bytes.toBytesBinary(colValTrim));
+        break;
+
+      case BOOLEAN:
+        recBuilder.set(fieldName, Boolean.parseBoolean(colValTrim));
+        break;
+
+      case STRING:
+        char[] colValArr = fieldValStr.toCharArray();
+        int length = fieldValStr.length();
+        // Right trim string
+        for (int i = length - 1; i > -1; i--) {
+          if (!Character.isWhitespace(colValArr[i])) {
+            break;
+          }
+          length--;
+        }
+        recBuilder.set(fieldName, fieldValStr.substring(0, length));
+        break;
+
+      case NULL:
+        recBuilder.set(fieldName, null);
+        break;
+
+      default:
+        // shouldn't ever get here
+        String err =
+          ResourceConstants.ERR_FIELD_VAL_CONVERT.getMsgForKeyWithCode(fieldName,
+                                                                       fieldValStr, "any Schema type");
+
+        throw new UnexpectedFormatException(err);
+    }
+  }
+
+  private void handleConversionException(String schemaTypeString, String fieldName, String fieldVal, Exception e)
+    throws IOException {
+
+    String err = ResourceConstants.ERR_FIELD_VAL_CONVERT.getMsgForKeyWithCode(fieldName, fieldVal, schemaTypeString);
+    throw new IOException(err, e);
+  }
+
+  /**
+   * Removes minus sign from the end of the string and puts it at the beginning.
+   * This handling is required for number values because some profile settings in
+   * Ariba may result in the minus sign being put at the end of the number like
+   * {@code 12345-}.
+   *
+   * @param fieldValTrim Trimmed field native value
+   * @return String with minus appended at the beginning if it was at the end.
+   * Else returns the same string
+   */
+  private String handleMinusAtEnd(String fieldValTrim) {
+    char lastChar = fieldValTrim.charAt(fieldValTrim.length() - 1);
+    if (lastChar == ResourceConstants.HYPHEN) {
+      StringBuilder sb = new StringBuilder(fieldValTrim);
+      sb.deleteCharAt(sb.length() - 1).insert(0, lastChar);
+      fieldValTrim = sb.toString();
+    }
+
+    return fieldValTrim;
+  }
+
+
+  @Nullable
+  protected LocalDate getSourceSpecificDateValue(String rawDateValue) {
```

Comment:
these methods should be private

Reply:
Fixed

Reply:
I don't see this fixed. Missing commit?

Reply:
This method is used for testing so made it protected again and added annotation @VisibleForTesting.
In this commit (https://github.com/data-integrations/ariba-plugins/pull/2/commits/4c63631ded44b52120d0ad7a20c370dc4ebd5f2b)

Reply:
Fixed